backend/src/routes/user.ts

```diff
@@ -148,10 +149,35 @@ router.get("/mentee/:userId", [verifyAuthToken], async (req: Request, res: Respo
       if (!mentee) {
         throw ServiceError.MENTEE_WAS_NOT_FOUND;
       }
+      const {
```

Comment:
This whole `if (role === "mentee") { ... } else { ... }` structure has some code duplicated between the two branches - consider moving the database query outside of the branches.

Reply:
done